src/components/NodeView.tsx

```diff
@@ -41,89 +40,85 @@ const NodeView: FC<Props> = ({
           >
-            <div
-              className={`node node-id=${id} ${feedbackClass}`}
-              // TODO: allow context menu on nodes?
-              // onContextMenu={handleContextMenu}
-              {...props}
-            >
-              {formulas.map((form, index) => {
-                return (
-                  <FormulaView
-                    key={`${form}-${index}`}
-                    node={node}
-                    index={index}
-                    dispatch={dispatch}
-                    {...form}
-                  />
-                )
+            <ArcherElement
+              id={id}
+              relations={forest.map((child) => {
+                return {
+                  targetId: child.id,
+                  targetAnchor: 'top',
+                  sourceAnchor: 'bottom',
+                }
               })}
+            >
+              <div
+                // TODO: allow context menu on nodes?
+                // onContextMenu={handleContextMenu}
+                style={{
+                  borderStyle: feedbackMap ? 'solid' : 'none',
+                  borderColor: feedbackClass === 'correct' ? 'green' : 'red',
+                  borderWidth: '1.5px',
+                }}
+                {...props}
+              >
+                {formulas.map((form, index) => {
+                  return (
+                    <FormulaView
+                      key={`${form}-${index}`}
+                      node={node}
+                      index={index}
+                      dispatch={dispatch}
+                      {...form}
+                    />
+                  )
+                })}
+              </div>
+            </ArcherElement>
+          </Tooltip>
+        </div>
 
-              {node.id !== '' ? (
-                <div className="justification">
-                  <AutoSizeInput
-                    className="rule"
-                    onChange={({ currentTarget: { value: rule } }) =>
-                      dispatch(updateJustification(firstRow(node), { rule }))
-                    }
-                    value={rule}
-                    placeholder="rule"
-                  />
-                  <AutoSizeInput
-                    className="row"
-                    onChange={({ currentTarget: { value: parentRow } }) =>
-                      dispatch(
-                        updateJustification(firstRow(node), { parentRow })
-                      )
-                    }
-                    value={parentRow}
-                    placeholder="row"
-                  />
-                </div>
-              ) : (
-                'AS'
-              )}
+        {forest.map((child, index) => {
+          return (
+            <div
+              key={child.id}
+              style={{
+                gridColumn: `${index} / span 1`,
+                gridRow: child.formulas[0]
+                  ? `${child.formulas[0].row - firstRow(node) + 1}`
```

Comment:
See comment in NodeViewBase.tsx.

Reply:
This is actually different, the plus one is needed to account for grid indexing starting at 1 (nextRow is only value that is +1 from what is actually needed, so in that specific case above can cancel out)

Reply:
You're right. I made the comment on the wrong line.